hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsRegistryImpl.java

```diff
@@ -283,16 +283,19 @@ public void discardMetrics(Object... objects) {
 
     List<ProbeInstance> getSortedProbeInstances() {
         long modCountLocal = modCount.get();
-        SortedProbeInstances sortedProbeInstances = this.sortedProbeInstances.get();
-        if (sortedProbeInstances.mod < modCountLocal) {
+        final SortedProbeInstances sortedInstancesOld = this.sortedProbeInstances.get();
+        final SortedProbeInstances sortedInstances;
+        if (sortedInstancesOld.mod < modCountLocal) {
             List<ProbeInstance> sorted = new ArrayList<>(probeInstances.values());
             Collections.sort(sorted, COMPARATOR);
-            sortedProbeInstances = new SortedProbeInstances(modCountLocal, sorted);
+            sortedInstances = new SortedProbeInstances(modCountLocal, sorted);
             // if some other thread sorted in the meantime, ignore our sorting
-            this.sortedProbeInstances.compareAndSet(sortedProbeInstances, sortedProbeInstances);
+            this.sortedProbeInstances.compareAndSet(sortedInstancesOld, sortedInstancesOld);
```

Comment:
I think the `update` value of the CAS should be `sortedInstances`.

Reply:
absolutely, fixed